### PR TITLE
Geoutils

### PR DESCRIPTION
**Pull request by pedroccufc**
## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have read the **CONTRIBUTING** document.
-   [x] My code follows the code style of this project (see `.code-style.md`).
-   [x] All new and existing tests passed (see `.testing.md`).
-   [x] I have added tests to cover my changes.
-   [x] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.

## Description

-   What is the current behavior? (You can also link to an open issue here)
    

-   What is the new behavior (if this is a feature change)?
    
The new GeoUtils module uses GeoPandas to perform geographic operations, being specific to this structure. For example, the module makes geometric transformations, to check if a point is inside or outside a polygon.

### CODE REVIEW

File: pymove/utils/geoutils.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    LATITUDE,
+    LATITUDE_DECODE,
+    LONGITUDE,
+    LONGITUDE_DECODE,
+)
+from pymove.utils.log import progress_bar
+
+
+def v_color(ob):
+    """
```
Comment (regispires) [inline pymove/utils/geoutils.py:18]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

File: pymove/utils/geoutils.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+def binary():
+    """
+    Transform base32 values into binary format.
+
+    Return
+    ------
+    array
+        binary form of each value in base32
+    """
+    bin = np.full(len(BASE_32), 0, dtype=np.ndarray)
```
Comment (regispires) [inline pymove/utils/geoutils.py:148]: Codacy found an issue: [Redefining built-in 'bin'](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

File: pymove/utils/geoutils.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        list('{0:05b}'.format(x)), dtype=int
+    ) for x in range(0, len(BASE_32))
+]
+
+
+BASE_32_TO_BIN = dict(zip(BASE_32, BINARY))
+
+
+def v_color(ob):
+    """
```
Comment (regispires) [inline pymove/utils/geoutils.py:27]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

Comment (regispires) [inline pymove/utils/geoutils.py:27]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

Comment (regispires) [inline pymove/utils/geoutils.py:27]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

File: pymove/utils/constants.py [hunk 4]
```diff
@@ -5,6 +5,23 @@
...
 
+GEOHASH = 'geohash'
+BIN_GEOHASH = 'bin_geohash'
+LATITUDE_DECODE = 'lat_d'
+LONGITUDE_DECODE = 'lon_d'
+
+COLOR = {
+    True: '#6699cc',
+    False: '#ffcc33'
+}
```
Comment (flych3r) [inline pymove/utils/constants.py:16]: Já tem um dicionário de cores. Pode colocar essas cores lá como chave tambem.

File: pymove/utils/geoutils.py [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    Returns '#ffcc33' if object crosses,
+    otherwise it returns '#6699cc'.
+
+    Parameters
+    ----------
+    ob : geometry object
+        Any geometric object
```
Comment (flych3r) [inline pymove/utils/geoutils.py:34]: Vale pro restante. Tem que ficar identado.
````python
"""
Parameters
----------
    nome: tipo
        descrição
"""
````

File: pymove/utils/constants.py [hunk 6]
```diff
@@ -5,6 +5,23 @@
 TID = 'tid'
 UID = 'user_id'
 
+GEOHASH = 'geohash'
+BIN_GEOHASH = 'bin_geohash'
+LATITUDE_DECODE = 'lat_d'
+LONGITUDE_DECODE = 'lon_d'
```
Comment (flych3r) [inline pymove/utils/constants.py:11]: Melhor chamar de `lat_decode`, só `lat_d` não da pra entender o que seria.

File: pymove/utils/geoutils.py [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+        list('{0:05b}'.format(x)), dtype=int
+    ) for x in range(0, len(BASE_32))
+]
+
+
+BASE_32_TO_BIN = dict(zip(BASE_32, BINARY))
+
+
+def v_color(ob):
+    """
```
Comment (regispires) [inline pymove/utils/geoutils.py:27]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

Comment (regispires) [inline pymove/utils/geoutils.py:27]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

Comment (regispires) [inline pymove/utils/geoutils.py:27]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

Comment (regispires) [inline pymove/utils/geoutils.py:27]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5971443)

**Pull request merged**